buildconfig/CMake/GNUSetup.cmake

```diff
@@ -115,3 +115,13 @@ if( CMAKE_COMPILER_IS_GNUCXX )
     set( CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS} -static-libgcc -static-libstdc++" )
   endif()
 endif()
+
+option (COLORED_COMPILER_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." TRUE)
+
+if(COLORED_COMPILER_OUTPUT)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    add_compile_options(-fdiagnostics-color=always)
```

Comment:
If the `COLORED_COMPILER_OUTPUT` is `False` should we set `-fdiagnostics-color=never`?

Reply:
Is an explicit else necessary? I think the flag being false will make it behave as it has until before this PR

Reply:
I was just wondering whether a developer would want `False` to mean definitely turn this off and not auto detect whether it should be on.

Reply:
That is true, I'll add the never for `False` as it makes sense